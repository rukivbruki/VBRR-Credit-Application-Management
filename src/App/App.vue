<template>
  <div>
    <el-container class="common-layout is-vertical">
      <Header></Header>
      <div>
        <Component :is="this.$route.meta.layout || 'div'">
          <router-view />
        </Component>
      </div>
      <Footer></Footer>
    </el-container>
  </div>
</template>

<script>
import Header from '../components/Header/Header';
import Footer from '../components/Footer/Footer';
import LeftSideMenu from '../components/MainMenu/LeftSideMenu/LeftSideMenu';
import SelectedContent from '../components/MainMenu/SelectedContent/Selected–°ontent';

export default {
  components: { SelectedContent, LeftSideMenu, Footer, Header },
  beforeMount() {
    this.$store.dispatch('setAllDictionaries');
  },
};
</script>

<style>
body > .el-container {
  margin-bottom: 40px;
}

.el-container:nth-child(5) .el-aside,
.el-container:nth-child(6) .el-aside {
  line-height: 260px;
}

.el-container:nth-child(7) .el-aside {
  line-height: 320px;
}

.common-layout {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}
</style>
