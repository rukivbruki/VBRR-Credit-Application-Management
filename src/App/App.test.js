import { mount } from '@vue/test-utils';
import App from './App';
describe('App.vue render', () => {
  test('renders correctly', () => {
    expect(mount(App).element).toMatchSnapshot();
  });
});
