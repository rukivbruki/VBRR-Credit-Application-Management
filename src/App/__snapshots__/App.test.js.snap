// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue render renders correctly 1`] = `
<div
  class="common-layout"
>
  <el-container
    class="is-vertical"
  >
    <div>
      <el-header>
        Header
      </el-header>
    </div>
    <el-container>
      <el-aside>
        <el-tabs
          stretch="true"
          style="height: 100%;"
          tab-position="left"
          type="border-card"
        >
          <el-tab-pane
            label="Пользователи"
          >
            Информация о пользователях 
          </el-tab-pane>
          <el-tab-pane
            label="Файлы"
          >
            Файлы и данные
          </el-tab-pane>
          <el-tab-pane
            label="Обслуживание"
          >
            Работа с клиентами
          </el-tab-pane>
          <el-tab-pane
            label="Кредитование"
          >
            Оформление заявок на кредитование 
          </el-tab-pane>
        </el-tabs>
      </el-aside>
      <el-main>
        
        <el-card
          class="box-card"
          shadow="hover"
        >
          
          <div
            class="text item"
          >
            Текущие задачи
          </div>
          <div
            class="text item"
          >
            Смена пароля
          </div>
          
        </el-card>
        <el-card
          class="box-card"
          shadow="hover"
        >
          
          <div
            class="text item"
          >
            Автоматизированные рабочие места
          </div>
          <div
            class="text item"
          >
            Филиалы/Точки обслуживания
          </div>
          <div
            class="text item"
          >
            Параметры точек обслуживания
          </div>
          
        </el-card>
        <el-card
          class="box-card"
          shadow="hover"
        >
          
          <div
            class="text item"
          >
            Дизайн бизнес процессов
          </div>
          <div
            class="text item"
          >
            Справочник массовой обработки
          </div>
          <div
            class="text item"
          >
            Дизайнер вызова по расписанию
          </div>
          
        </el-card>
        
      </el-main>
    </el-container>
    <div>
      <el-footer>
        Footer
      </el-footer>
    </div>
  </el-container>
</div>
`;
